cmd/algorand-indexer/daemon.go

```diff
@@ -49,50 +52,135 @@ var (
 	defaultApplicationsLimit  uint32
 	enableAllParameters       bool
 	indexerDataDir            string
+	cpuProfile                string
+	pidFilePath               string
+	configFile                string
 )
 
 var daemonCmd = &cobra.Command{
 	Use:   "daemon",
 	Short: "run indexer daemon",
 	Long:  "run indexer daemon. Serve api on HTTP.",
-	//Args:
 	Run: func(cmd *cobra.Command, args []string) {
 		var err error
 		config.BindFlags(cmd)
 		err = configureLogger()
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
-			os.Exit(1)
+			panic(exit{1})
+		}
+
+		if indexerDataDir == "" {
+			indexerDataDir = os.Getenv("INDEXER_DATA")
+		}
+
+		if indexerDataDir == "" {
+			fmt.Fprint(os.Stderr, "indexer data directory was not provided")
+			panic(exit{1})
+		}
+
+		// Create the directory if it doesn't exist
+		if _, err := os.Stat(indexerDataDir); os.IsNotExist(err) {
+			err := os.Mkdir(indexerDataDir, 0755)
+			maybeFail(err, "indexer data directory error, %v", err)
+		}
+
+		// Detect the various auto-loading configs from data directory
+		indexerConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
+		paramConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadParameterConfigName))
+
+		// If we auto-loaded configs but a user supplied them as well, we have an error
+		if indexerConfigFound {
+			if configFile != "" {
+				logger.Errorf(
+					"indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one.",
+					filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
+				panic(exit{1})
```

Comment:
Instead of a panic, could we return for these?
```suggestion
				return
```

Reply:
Don't we want to return an error code since this would be an invalid configuration?  It would help with scripts who look at the return code of a program.

Reply:
You're right. We could change this function to return an `error` and use more conventional error handling though. I'll let you decide.

Reply:
I think consistency is important, it is probably easier to just have this panic out that way we aren't dealing with the "should I return an error or panic here" train of thought.